wallet/index.js

```diff
@@ -73,42 +75,45 @@ class Wallet {
     constructor(ctx) {
         this.ctx = ctx;
         this.log = ctx.log.child({module: 'Wallet'});
-        this.config = ctx.config.client.wallet;
-
+        this.keystorePath = path.join(config.get('datadir'), config.get('wallet.keystore_path'));
         // Events
         // transactionEventEmitter emits the TRANSACTION_EVENT type
         this.transactionEventEmitter = new events.EventEmitter();
     }
 
     async start() {
-        this.keystore_path = path.join(this.ctx.datadir, this.config.keystore_path);
-        if (!fs.existsSync(this.keystore_path)) {
-            fs.mkdirSync(this.keystore_path, {recursive: true});
+        await makeSurePathExistsAsync(this.keystorePath);
+        try {
+            this.#secretPhrase = JSON.parse(
```

Comment:
It'd be much simpler to write `this.#secretPhrase = require(path.join(this.keystorePath, 'key.json')).phrase;` instead.